File: src/main/java/org/mitallast/queue/crdt/routing/allocation/AllocationStrategy.java
package org.mitallast.queue.crdt.routing.allocation;

import javaslang.control.Option;
import org.mitallast.queue.common.codec.Message;
import org.mitallast.queue.crdt.routing.RoutingTable;

public interface AllocationStrategy {

    Option<Message> update(RoutingTable routingTable);
}

File: src/main/java/org/mitallast/queue/crdt/routing/allocation/AllocationStrategy.kt
package org.mitallast.queue.crdt.routing.allocation

import io.vavr.control.Option
import org.mitallast.queue.common.codec.Message
import org.mitallast.queue.crdt.routing.RoutingTable

interface AllocationStrategy {

    fun update(routingTable: RoutingTable): Option<Message>
}
